### ISSUE TITLE
Source-build leg in runtime failing due to missing intermediates

### ISSUE THREAD

**Issue by crummel** [dotnet/source-build#3219]
Related issue: https://github.com/dotnet/runtime/issues/80284.

The banana-rid source-build leg in runtime is failing due to a missing runtime package:

```
  /__w/1/s/.dotnet/sdk/7.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1094: Unable to optimize assemblies for performance: a valid runtime package was not found. Either set the PublishReadyToRun property to false, or use a supported runtime identifier when publishing. When targeting .NET 6 or higher, make sure to restore packages with the PublishReadyToRun property set to true. [/__w/1/s/artifacts/source-build/self/src/Build.proj]
```

**Comment by omajid** [comment]
FYI @tmds

**Comment by tmds** [comment]
I've commented on the referenced issue.

**Comment by omajid** [comment]
Thanks, @tmds!

**Comment by crummel** [comment]
Appears to have been fixed by the uptake of the 7.0.101 SDK, see https://dev.azure.com/dnceng-public/public/_build/results?buildId=197994&view=logs&jobId=000ba8e3-34f8-51d2-8d06-9b61f10256d0&j=000ba8e3-34f8-51d2-8d06-9b61f10256d0&t=ca9b6e26-8c16-57f9-673e-2bc0dc655f25 for an example.
